### ISSUE TITLE
Emscripten compilation error Ubuntu 22.04

### ISSUE THREAD

**Issue by pierricgimmig** [beaufortfrancois/webgpu-cross-platform-app#3]
Running `emcmake cmake -B build-web && cmake --build build-web -j4` on Ubuntu 22.04 gives:

```
pierric@squid:~/git/webgpu-cross-platform-app$ emcmake cmake -B build-web && cmake --build build-web -j4
configure: cmake -B build-web -DCMAKE_TOOLCHAIN_FILE=/usr/share/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/usr/bin/node;--experimental-wasm-threads
-- Configuring done (1.3s)
-- Generating done (0.0s)
-- Build files have been written to: /home/pierric/git/webgpu-cross-platform-app/build-web
[ 50%] Building CXX object CMakeFiles/app.dir/main.cpp.o
/home/pierric/git/webgpu-cross-platform-app/main.cpp:35:11: error: use of undeclared identifier 'exit'
          exit(0);
          ^
/home/pierric/git/webgpu-cross-platform-app/main.cpp:63:12: error: no member named 'code' in 'wgpu::ShaderModuleWGSLDescriptor'
  wgslDesc.code = shaderCode;
  ~~~~~~~~ ^
/home/pierric/git/webgpu-cross-platform-app/main.cpp:97:8: error: no member named 'End' in 'wgpu::RenderPassEncoder'
  pass.End();
  ~~~~ ^
3 errors generated.
em++: error: '/usr/bin/clang++-13 -target wasm32-unknown-emscripten -DEMSCRIPTEN -D__EMSCRIPTEN_major__=3 -D__EMSCRIPTEN_minor__=1 -D__EMSCRIPTEN_tiny__=5 -fignore-exceptions -fvisibility=default -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -D_LIBCPP_ABI_VERSION=2 -fno-threadsafe-statics -Dunix -D__unix -D__unix__ -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/SDL --sysroot=/usr/share/emscripten/cache/sysroot -Xclang -iwithsysroot/include/compat -std=gnu++20 -MD -MT CMakeFiles/app.dir/main.cpp.o -MF CMakeFiles/app.dir/main.cpp.o.d -c /home/pierric/git/webgpu-cross-platform-app/main.cpp -o CMakeFiles/app.dir/main.cpp.o' failed (returned 1)
gmake[2]: *** [CMakeFiles/app.dir/build.make:76: CMakeFiles/app.dir/main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/app.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

```

**Comment by beaufortfrancois** [comment]
Re: `error: no member named 'code' in 'wgpu::ShaderModuleWGSLDescriptor'`, I suspect you're not using the last version of Emscripten. I use version 3.1.46.

**Comment by pierricgimmig** [comment]
Thnaks for the suggestion, I removed the build-web folder, tried again, and got:

```
app.js:6572 Failed to create WebGPU Context Provider
```

But this was because I hadn't enabled WebGpu in Chrome. 

Running `google-chrome-unstable --enable-unsafe-webgpu --use-vulkan=true --test-type --enable-features=Vulkan` finally gave me that red triangle I was looking for :-)

**Comment by beaufortfrancois** [comment]
Woohoo!

**Comment by beaufortfrancois** [comment]
Can you summarize all the steps you made to the original instructions so that others can benefit from you experience on Linux?

**Comment by pierricgimmig** [comment]
Sure! Here's what I did on Ubuntu 22.04:

- Download and install the latest Emscripten from https://emscripten.org/docs/getting_started/downloads.html, (3.1.47 for me)
- Install unstable Google Chrome through `sudo apt install google-chrome-unstable` for WebGpu support
- Build this project with `emcmake cmake -B build-web && cmake --build build-web -j4`
- In a terminal, at the root of this project, run `npx http-server` (you might have to install npm)
- Launch WebGpu-enabled unstable Google Chrome: `google-chrome-unstable --enable-unsafe-webgpu --use-vulkan=true --test-type --enable-features=Vulkan`
- Go to `http://127.0.0.1:8080/build-web/app.html` in the unstable browser

![image](https://github.com/beaufortfrancois/webgpu-cross-platform-app/assets/3687222/6abd8c57-911b-4130-b150-a34aa938c7c2)

**Comment by munrocket** [comment]
Faced with this problem too on Linux.

**Comment by beaufortfrancois** [comment]
Thank you @pierricgimmig for documenting your steps.
Hopefully it will help others in the future.
